### DISCUSSION TITLE
How to Apply a Globally Valid Set of Cuts After Root Cut Generation

### DISCUSSION THREAD

**Discussion by spkelle2** [coin-or/Cbc#598] [Q&A] [answered]
I have a set of cuts that I know are valid for a MILP. I would like to add them to the pool of cuts that are considered for every node in the branch and bound tree, and I would like to do so after cut generation at the root node has completed. How can I do so?

**Comment by jjhforrest** [comment, marked as answer]
You want to create a cut generator from CglStored and add that to the model.  Look at Cbc/examples/allCuts.cpp.  This may be out of date but will get you started.  You may be better off using a more flexible example e.g. driver6.cpp and using a call back to add the generator.

**Reply by spkelle2** [reply]
I'll have a look and write back my solution for anyone curious in the future. Thank you for the pointer!

**Reply by spkelle2** [reply]
This worked like a charm! Thank you!
